code/game/objects/structures/janicart.dm

```diff
@@ -1,145 +1,231 @@
+#define CART_HAS_MINIMUM_REAGENT_VOLUME !(reagents.total_volume < 1)
+
 /obj/structure/janitorialcart
 	name = "janitorial cart"
 	desc = "This is the alpha and omega of sanitation."
 	icon = 'icons/obj/janitor.dmi'
 	icon_state = "cart"
 	anchored = FALSE
 	density = TRUE
-	//copypaste sorry
 	var/amount_per_transfer_from_this = 5 //shit I dunno, adding this so syringes stop runtime erroring. --NeoFite
 	var/obj/item/storage/bag/trash/mybag
 	var/obj/item/mop/mymop
 	var/obj/item/pushbroom/mybroom
 	var/obj/item/reagent_containers/spray/cleaner/myspray
 	var/obj/item/lightreplacer/myreplacer
-	var/signs = 0
+	var/list/obj/item/clothing/suit/caution/held_signs = list()
 	var/max_signs = 4
 
-
 /obj/structure/janitorialcart/Initialize(mapload)
 	. = ..()
 	create_reagents(100, OPENCONTAINER)
 	GLOB.janitor_devices += src
 
+	register_context() //Context sensitive tooltips. See add_context()
+
 /obj/structure/janitorialcart/Destroy()
 	GLOB.janitor_devices -= src
+	QDEL_NULL(myreplacer)
+	QDEL_NULL(myspray)
+	QDEL_NULL(mybroom)
+	QDEL_NULL(mymop)
+	QDEL_NULL(mybag)
+	QDEL_LIST(held_signs)
 	return ..()
 
-/obj/structure/janitorialcart/examine(mob/user)
-	. = ..()
-	if(mymop)
-		. += span_info("<b>Right-click</b> to quickly remove [mymop].")
-	if(reagents.total_volume > 1)
-		. += span_info("<b>Right-click</b> with a mop to wet it.")
-		. += span_info("<b>Crowbar</b> it to empty it onto [get_turf(src)].")
-	if(mybag)
-		. += span_info("<b>Right-click</b> with an object to put it in [mybag].")
+/obj/structure/janitorialcart/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(istype(arrived, /obj/item/storage/bag/trash))
+		mybag = arrived
+	else if(istype(arrived, /obj/item/mop))
```

Comment:
should we really use else if in byond? is a switch possible here?

Reply:
Not supported.
http://www.byond.com/forum/post/2267853

Reply:
switches do a direct comparison against the value of the input.
ie

```
	switch(istype(arrived, /obj/item/mop))
		if(TRUE)
			DO THING
		if(FALSE)
			DO OTHER THING
		if(/obj/item/storage/bag/trash)
			this won't happen cause istype(arrived) is a true or false. 


				would be equivalent to writing out
			if(istype(arrived, /obj/item/mop) == /obj/item/storage/bag/trash)
```
			
DM doesn't support switches being used as arguments, so this is sadly the best we can do.